src/bst/core.clj

```diff
@@ -67,18 +67,21 @@
   (if (empty? tree) false
       (and (is-right-case? tree) (> (factor (:right tree)) 0))))
 
+(def update-height #(assoc %1 :height (+ 1 (max (height (:left %1)) (height (:right %1))))))
+
 (defn rotate-left
   "Returns the left rotated tree "
   [{:keys [root left right] :as tree}]
   (let [pivot (:root right)
         left-pivot (:left right)
         right-pivot (:right right)]
     (if (empty? tree) {}
-        {:root pivot
-         :left {:root root
-                :left left
-                :right left-pivot}
-         :right right-pivot})))
+        (update-height {:root pivot
```

Comment:
We should rename `:root` to `:data`, since it is storing data. A `:root` is an entire node, not just a scalar datum.